main/SAPI.c

```diff
@@ -549,17 +549,26 @@ SAPI_API void sapi_initialize_empty_request(void)
 
 static int sapi_extract_response_code(const char *header_line)
 {
-	int code = 200;
+	zend_ulong code = 200;
 	const char *ptr;
 
 	for (ptr = header_line; *ptr; ptr++) {
 		if (*ptr == ' ' && *(ptr + 1) != ' ') {
-			code = atoi(ptr + 1);
+			const char *code_begin = ptr + 1;
+			char *code_end;
+
+			code = ZEND_STRTOUL(code_begin, &code_end, 10);
+
+			/* rfc7230 3.1.2 status-code = 3DIGIT */
+			if(*code_begin < '0' || *code_begin > '9' || (code_end - code_begin) != 3) {
```

Comment:
if you are already doing this, it might be worth no to allow anything above 599 and also first check should be probably `*code_begin < '1'`.

also small coding style nit: there should be a space after `if`

Reply:
What I mean is basically that rfc7230 6 defines semantics and doesn't really allow anything else so might make sense to check that as well. Not sure how could be useful to allow 000 for example...

Reply:
Yes, actually my first patch also it's in [100, 999] but I removed because of my confusion. Since some of applications use 000 or 999 for their special kind of response, I confused it's under standard.

However, now I'm totally agree with it should be restricted in [100, 599] and there should be a space after if :)

Reply:
I made a commit :)